### PR TITLE
RavenDB-22426 Replace System.Data.SqlClient with Microsoft.Data.SqlClient

### PR DESCRIPTION
**Pull request by Lwiel**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22426/Replace-System.Data.SqlClient-NuGet-package-with-Microsoft.Data.SqlClient

### Additional description

Replaced `System.Data.SqlClient` package with `Microsoft.Data.SqlClient`

### Type of change

- [x] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [x] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [ ] No UI work is needed

### CODE REVIEW

File: src/Raven.Client/Documents/Operations/ETL/SQL/SqlProviderParser.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@ public static SqlProvider GetSupportedProvider(string factoryName)
         {
             switch (factoryName)
             {
-                case "System.Data.SqlClient":
+                case "Microsoft.Data.SqlClient":
```
Comment (arekpalinski) [inline src/Raven.Client/Documents/Operations/ETL/SQL/SqlProviderParser.cs:12]: We must not change that. Existing connection strings will have `"System.Data.SqlClient"`. We should just add `"Microsoft.Data.SqlClient"`

File: src/Raven.Server/Documents/ETL/Providers/SQL/RelationalWriters/RelationalDatabaseWriterBase.cs [hunk 2]
```diff
@@ -8,7 +8,7 @@ public class RelationalDatabaseWriterBase
 
         private static readonly string[] SqlServerFactoryNames =
         {
-            "System.Data.SqlClient",
+            "Microsoft.Data.SqlClient",
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/ETL/Providers/SQL/RelationalWriters/RelationalDatabaseWriterBase.cs:12]: Same as above

File: src/Raven.Studio/mockups/src/partials/modals/newDatabase.html [hunk 3]
```diff
@@ -352,7 +352,7 @@ <h3 class="panel-title">
                                                         <span class="caret"></span>
                                                     </button>
                                                     <ul class="dropdown-menu">
-                                                        <li><a href="#" value="System.Data.SqlClient">System.Data.SqlClient</a></li>
+                                                        <li><a href="#" value="Microsoft.Data.SqlClient">Microsoft.Data.SqlClient</a></li>
```
Comment (arekpalinski) [inline src/Raven.Studio/mockups/src/partials/modals/newDatabase.html]: This is in `mockups` dir so I believe it's okay. CC @ml054

File: src/Raven.Studio/typescript/models/database/settings/connectionStringSqlEtlModel.ts [hunk 4]
```diff
@@ -9,7 +9,7 @@ import sqlConnectionStringSyntax = require("common/helpers/database/sqlConnectio
 class connectionStringSqlEtlModel extends connectionStringModel {
 
     static sqlProviders: Array<valueAndLabelItem<string, string> & { tooltip: string }> = [
-        { value: "System.Data.SqlClient", label: "Microsoft SQL Server", tooltip: sqlConnectionStringSyntax.mssqlSyntax },
+        { value: "Microsoft.Data.SqlClient", label: "Microsoft SQL Server", tooltip: sqlConnectionStringSyntax.mssqlSyntax },
```
Comment (arekpalinski) [inline src/Raven.Studio/typescript/models/database/settings/connectionStringSqlEtlModel.ts:12]: I think that we need to leave the old entry for backward compatibility and add a new one. CC @ml054 

Also see https://issues.hibernatingrhinos.com/issue/RavenDB-22365/Confusing-message-for-SQL-ETL-after-upgrade-from-5.4.115-to-118

File: src/Raven.Studio/typescript/models/database/tasks/sql/sqlMigration.ts [hunk 5]
```diff
@@ -305,7 +305,7 @@ class sqlMigration {
             case "MySQL_MySqlConnector":
                 return "MySqlConnector.MySqlConnectorFactory";
             case "MsSQL":
-                return "System.Data.SqlClient";
+                return "Microsoft.Data.SqlClient";
```
Comment (arekpalinski) [inline src/Raven.Studio/typescript/models/database/tasks/sql/sqlMigration.ts:308]: Same issue as in other places

File: src/Raven.Client/Documents/Operations/ETL/SQL/SqlProviderParser.cs (hunk 1 above)
Comment (Lwiel) [reply src/Raven.Client/Documents/Operations/ETL/SQL/SqlProviderParser.cs:12]: Fixed

File: src/Raven.Server/Documents/ETL/Providers/SQL/RelationalWriters/RelationalDatabaseWriterBase.cs (hunk 2 above)
Comment (Lwiel) [reply src/Raven.Server/Documents/ETL/Providers/SQL/RelationalWriters/RelationalDatabaseWriterBase.cs:12]: Fixed

**Review by arekpalinski — APPROVED** [review]

File: src/Raven.Server/Documents/ETL/Providers/SQL/RelationalWriters/SqlConnectionStringUtil.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+namespace Raven.Server.Documents.ETL.Providers.SQL.RelationalWriters;
+
+public static class SqlConnectionStringUtil
+{
+    public static string GetConnectionStringWithOptionalEncrypt(string connectionString)
+    {
+        if (connectionString.EndsWith(';') == false)
```
Comment (ppekrol) [inline src/Raven.Server/Documents/ETL/Providers/SQL/RelationalWriters/SqlConnectionStringUtil.cs:9]: maybe add 
`connectionString = connectionString.Trim();` in first line of this method?

File: src/Raven.Studio/typescript/models/database/tasks/ongoingTaskSqlEtlEditModel.ts [hunk 7]
```diff
@@ -127,15 +127,13 @@ class ongoingTaskSqlEtlEditModel extends ongoingTaskEditModel {
             Disabled: this.taskState() === "Disabled",
             MentorNode: this.manualChooseMentor() ? this.mentorNode() : undefined,
             PinToMentorNode: this.pinMentorNode(),
-            FactoryName: "System.Data.SqlClient",
```
Comment (ppekrol) [inline src/Raven.Studio/typescript/models/database/tasks/ongoingTaskSqlEtlEditModel.ts:130]: don't we need to put here `Microsoft.Data.SqlClient` ?

Comment (ml054) [reply src/Raven.Studio/typescript/models/database/tasks/ongoingTaskSqlEtlEditModel.ts:130]: no

File: src/Raven.Server/Documents/ETL/Providers/SQL/RelationalWriters/SqlConnectionStringUtil.cs (hunk 6 above)
Comment (Lwiel) [reply src/Raven.Server/Documents/ETL/Providers/SQL/RelationalWriters/SqlConnectionStringUtil.cs:9]: Added

**Pull request merged**